Homely - Front-End

Wahaj-Ali opened the pull request:
Hello 👋 

We are a team of 4 members, but we are divided into 2 subteams to handle the front end and the back end of our final Capstone project.

I, @Wahaj-Ali, and @blase147 worked on this repo(Front-End). While @akhror-valiev & @belmeetmule worked on the [back-end](https://github.com/belmeetmule/homely-backend).

## HIGHLIGHTS
#### In this PR
- Add a sidebar with the links to pages such as Houses (theme of our project), Add House, Delete House, My Reservations, and Login/ Logout(as required).
- Add House component has a form that lets the user add a new house to the list of houses.
- Delete House component lets the user delete a house from the list.
- From the list of the houses(Main page of the App), the user is able to navigate to each house to see its details along with the option to book his/her visit(The user is only able to do so if he/she is logged in).
- The login button takes the user to a form where the user can enter his email along with a password to sign in.
- In order to Sign up, the user can find a link from the login page that takes the user to the signup form. It requires details such as full name, email, and password.
- If the user is logged in, only then the user can navigate to the "My Reservations" page. This page consists of all the bookings that the user made.
- Use "sliderjs" for the layout of the main page to match the design provided.
- Add responsive design for the App to adjust to the mobile view.
- Use Redux toolkit for the state management of the App and Axios for making HTTP requests to the backend server.
- Add an additional feature to Implement proper user authentication from the front-end to the server.
- Add tests for the components.

[Link](https://github.com/belmeetmule/homely-backend/pull/28) to the back-end PR.
[Kanban Board](https://github.com/belmeetmule/homely-backend/projects/1).

Wahaj-Ali committed:
Set up React App

In this commit
- Install React App.
- Remove unnecessary files and lines from the code.

blase147 committed:
Install linters

blase147 committed:
Install react-router-dom

blase147 committed:
Install react-redux

blase147 committed:
Update Readme, fix linters errors

blase147 committed:
Update Readme

blase147 committed:
Update Readme

blase147 committed:
Merge pull request #1 from Wahaj-Ali/react-setup

React setup

Wahaj-Ali committed:
Remove linter errors.

Wahaj-Ali committed:
Merge pull request #2 from Wahaj-Ali/react-setup

Remove linter errors.

Wahaj-Ali committed:
Install react-icons

Wahaj-Ali committed:
Create SideBar.js

Wahaj-Ali committed:
Create SideBar.css

Wahaj-Ali committed:
Update App.js

Wahaj-Ali committed:
Install Sass

Wahaj-Ali committed:
Update SideBar.css

Wahaj-Ali committed:
Update App.js

Wahaj-Ali committed:
Create _config.scss

Wahaj-Ali committed:
Create Houses.js

Add houses component which shows the list of all available houses to be booked.

Wahaj-Ali committed:
Create Houses.scss

Add styles to the houses component

Wahaj-Ali committed:
Update SideBar.css

Wahaj-Ali committed:
Install Swiper

Wahaj-Ali committed:
Update Houses.js

Use Swiper to implement the slide functionality for the main page.

Wahaj-Ali committed:
Update Houses.scss

Update styles for the swiper next and prev arrows.

Wahaj-Ali committed:
Update SideBar.css

Wahaj-Ali committed:
Made 'Houses' page as the main page of the App.

Wahaj-Ali committed:
Update SideBar.css

Wahaj-Ali committed:
Update Houses.js

Edit dummy description for a house.

Wahaj-Ali committed:
Update Houses.scss

Wahaj-Ali committed:
Update _config.scss

Wahaj-Ali committed:
Update SideBar.js

Change a tag with the Link to make the App SPA. Now the App does not reloads when navigating to another page.

Wahaj-Ali committed:
Update SideBar.css

Remove linter errors.

Wahaj-Ali committed:
Merge pull request #3 from Wahaj-Ali/home

 Main page(Frontend)

Wahaj-Ali committed:
Update Houses.js

Remove mock data from the component.

Wahaj-Ali committed:
Create mockData.js

Wahaj-Ali committed:
Create HousesDetails.js

Add a new component to display the details of each item(house).

Wahaj-Ali committed:
Update App.js

Update routes for the item-details page.

Wahaj-Ali committed:
Update mockData.js

Wahaj-Ali committed:
Create icon.png

Wahaj-Ali committed:
Remove the image hover property from the config file and add it to the houses.scss

Wahaj-Ali committed:
Create HouseDetails component.

Wahaj-Ali committed:
Create HouseDetails.scss

Add styles for the House Details component.

Wahaj-Ali committed:
Update App.js

Wahaj-Ali committed:
Update SideBar.js

Remove the menu button and its functionality to slide/hide the sidebar.

Wahaj-Ali committed:
Update SideBar.css

Wahaj-Ali committed:
Update Houses.js

Add class to the section used in the houses component.

Wahaj-Ali committed:
Update Houses.scss

Adjust the positioning of the houses component section in accordance with the sidebar's permanent visibility.

Wahaj-Ali committed:
Remove linter errors.

Wahaj-Ali committed:
Update HouseDetails.js

Add 'Book/reserve house' button.

Wahaj-Ali committed:
Update HouseDetails.scss

Add styles to the 'Book House' button.

Wahaj-Ali committed:
Change the background to a lighter color for the alternative p tags in the House Details page.

Wahaj-Ali committed:
Remove linte rerrors.

Wahaj-Ali committed:
Update styles for the Details page.

Wahaj-Ali committed:
Merge pull request #4 from Wahaj-Ali/item-details

Details page(Frontend)

Wahaj-Ali committed:
Update SideBar.js

Edit sidebar navigation to include Log in access link & remove 'reserve' link.

Wahaj-Ali committed:
Update App.js

Update the routes so that the sidebar is only available for the components it is required for excluding the log in page.

Wahaj-Ali committed:
Create Login.js

Wahaj-Ali committed:
Install Redux, redux-thunk, redux-toolkit and axios.

Wahaj-Ali committed:
Create loginSlice.js

Wahaj-Ali committed:
Create store.js

Wahaj-Ali committed:
Update App.js

Wahaj-Ali committed:
Update Login.js

Wahaj-Ali committed:
Update Login.js

Use useSelector to access the state from the auth user to check if the user is logged in. If logged in, then redirect the user to the homepage.

Wahaj-Ali committed:
Update SideBar.js

Update sidebar to include sign out button and exclude the link to login form as soon as the user is logged in and vice versa.

Wahaj-Ali committed:
Delete loginSlice.js

Wahaj-Ali committed:
Create authSlice.js

Wahaj-Ali committed:
Update store.js

Wahaj-Ali committed:
Update SideBar.js

Now the 'My reservations' is only shown to the logged in users.

Wahaj-Ali committed:
Update Login.js

Earlier, upon reloading, the user automatically logged out.  Use useSelector hook to check if the user is logged in along with the token availibilty. So now the user does not log out on reloading the app.

Wahaj-Ali committed:
Update SideBar.js

Use token for conditional rendering of the some of the elements in the Sidebar component.

Wahaj-Ali committed:
Update authSlice.js

Add action to log out the current user.

Wahaj-Ali committed:
Update Login.js

Wahaj-Ali committed:
Update SideBar.js

Wahaj-Ali committed:
Update authSlice.js

Set state.loggedIn to false when user successfully logs out.

Wahaj-Ali committed:
Update Login.js

Comment out console.log statements.

Wahaj-Ali committed:
Update SideBar.js

Remove console log statements.

Wahaj-Ali committed:
Update SideBar.js

Wahaj-Ali committed:
Create Login.scss

Add styles for the login page.

Wahaj-Ali committed:
Update Login.scss

remove linter errors.

Wahaj-Ali committed:
Update Login.js

Wahaj-Ali committed:
Modify style linters.

Wahaj-Ali committed:
Update Login.js

Add a button to head back to the main-page of the app from the login page.

Wahaj-Ali committed:
Update Login.scss

Add styles to the log in button.

Wahaj-Ali committed:
Update Login.scss

Remove commented lines.

Wahaj-Ali committed:
Update SideBar.js

Wahaj-Ali committed:
Create houseSlice.js

Use the API endpoint to fetch all the houses and their details.

Wahaj-Ali committed:
Update Houses.js

Display the list of houses (from API endpoint).

Wahaj-Ali committed:
Update store.js

Wahaj-Ali committed:
Update Houses.js

Show loading when data is being fetched from the API.

Wahaj-Ali committed:
Install prop-types

Wahaj-Ali committed:
Create houseDetailsSlice.js

Add a house slice to fetch the data for a particular house from the API.

Wahaj-Ali committed:
Update HouseDetails.js

Use dispatch of implement the action of fetching house details.

Wahaj-Ali committed:
Update store.js

Add house reducer in the store.

Wahaj-Ali committed:
Update HouseDetails.scss

Wahaj-Ali committed:
Update HouseDetails.js

Wahaj-Ali committed:
Update HouseDetails.scss

Wahaj-Ali committed:
Remove styleint errors.

Wahaj-Ali committed:
Create Register.js

Add the sign-up component.

Wahaj-Ali committed:
Update App.js

Add a route to the registration page to add a new user.

Wahaj-Ali committed:
Update HouseDetails.scss

Remove linter errors.

Wahaj-Ali committed:
Create regSlice.js

Wahaj-Ali committed:
Update store.js

Wahaj-Ali committed:
Update Register.js

Wahaj-Ali committed:
Update Register.js

Render a message on successful and unsuccessful sign up.

Wahaj-Ali committed:
Update Register.js

Add useEffect hook to redirect to login page after successful sign up.

Wahaj-Ali committed:
Update Register.js

Use IsSignedUp to make the user to be able to come back to the sign up page after being redirected to the login page after successful sign up.

Wahaj-Ali committed:
Update regSlice.js

Wahaj-Ali committed:
Merge pull request #7 from Wahaj-Ali/auth

Authentication & House list and details.

Wahaj-Ali committed:
Update SideBar.js

Handle the login functionality, when a user reopens the app without logging out, he can now logout from the new session.

Wahaj-Ali committed:
Create addHouse.js

Wahaj-Ali committed:
Update App.js

Add navigation link to the add house form.

Wahaj-Ali committed:
Create addHouseSlice.js

Add an action to add house using the API endpoint.

Wahaj-Ali committed:
Update store.js

Wahaj-Ali committed:
Update addHouse.js

Add controls for the labels & dispatch the add house action when the user submits the form.

Wahaj-Ali committed:
Update addHouse.js

Wahaj-Ali committed:
Create addHouse.scss

Wahaj-Ali committed:
Update Login.scss

Wahaj-Ali committed:
Update addHouse.js

Wahaj-Ali committed:
Update addHouse.scss

Wahaj-Ali committed:
Update App.js

Update route for add house component. Now when a user opens the form to add a new house, the sidebar disappears.

Wahaj-Ali committed:
Update addHouse.js

Add a button in the add house form to take the user back to the homepage of the App.

Wahaj-Ali committed:
Update Login.scss

Change the position of the login button.

Wahaj-Ali committed:
Update Register.js

Add a button to take the user back to the main page of the App.

Wahaj-Ali committed:
Update Login.scss

Wahaj-Ali committed:
Update addHouse.scss

Remove linter errors.

Wahaj-Ali committed:
Merge pull request #8 from Wahaj-Ali/add-house

Add house link

Wahaj-Ali committed:
Update addHouse.js

Wahaj-Ali committed:
Update addHouse.js

Upon successful submission of the add-house form, user is redirected to the home page.

Wahaj-Ali committed:
Merge pull request #9 from Wahaj-Ali/add-house

Add house

Wahaj-Ali committed:
Update HouseDetails.js

Upon clicking 'Book your visit' button, the user is redirected to a new reservation page to confirm the reservation/appointment.

Wahaj-Ali committed:
Update App.js

Add a route to the reservation confirmation page.

Wahaj-Ali committed:
Update HouseDetails.scss

Wahaj-Ali committed:
Create reservationsSlice.js

Wahaj-Ali committed:
Update store.js

Add reservation reducer in the Redux store.

Wahaj-Ali committed:
Update App.js

blase147 committed:
Create deleteHouse comontnt & deleteHouse slice

blase147 committed:
Fix linter errors

Wahaj-Ali committed:
Update authSlice.js

Update the authSlice to get the information of the current user from the API endpoint.

Wahaj-Ali committed:
Update reservationsSlice.js

blase147 committed:
Update deleteHouse

Wahaj-Ali committed:
Install datepicker.

Wahaj-Ali committed:
Update reservationsSlice.js

Wahaj-Ali committed:
Update App.js

Wahaj-Ali committed:
Create Reservation.js

Wahaj-Ali committed:
Update Reservation.js

Wahaj-Ali committed:
Create Reservation.scss

Wahaj-Ali committed:
Update Reservation.scss

Remove unnecessary code.

Wahaj-Ali committed:
Update Reservation.js

Set the reservation date to be provided to the API as the selected date using the date provider in React.

Wahaj-Ali committed:
Update Reservation.scss

Wahaj-Ali committed:
Update Reservation.js

Wahaj-Ali committed:
Update Reservation.scss

Remove linter errors.

Wahaj-Ali committed:
Create MyReservations.js

Wahaj-Ali committed:
Update App.js

blase147 committed:
Fix linter errors

blase147 committed:
Fix linter errors

blase147 committed:
Update deleteHouse component

blase147 committed:
Update deleteHouse component

blase147 committed:
Update deleteHouse component

Wahaj-Ali committed:
Update reservationsSlice.js

Wahaj-Ali committed:
Update MyReservations.js

Wahaj-Ali committed:
Update MyReservations.js

Wahaj-Ali committed:
Update MyReservations.js

Wahaj-Ali committed:
Create MyReservations.scss

Wahaj-Ali committed:
Update MyReservations.js

My reservation page now contains the name of each booked house.

Wahaj-Ali committed:
Update MyReservations.js

Wahaj-Ali committed:
Update MyReservations.scss

Wahaj-Ali committed:
Update Reservation.js

Upon successful confirmation of the visit(when user clicks the confirm visit button), the app redirects to the my reservations page.

Wahaj-Ali committed:
Remove Linter Errors.

Wahaj-Ali committed:
Update the styles for the home page to adjust for a mobile screen, making the design responsive.

In this commit
- Use media query to update the styles to adjust to a mobile screen.

Wahaj-Ali committed:
Merge pull request #11 from Wahaj-Ali/my-reservations-wahaj

My reservations

Wahaj-Ali committed:
Update HouseDetails.scss

Use media query to update styles for a responsive design of the component.

Wahaj-Ali committed:
Remove linter errors.

Wahaj-Ali committed:
Update Houses.scss

Wahaj-Ali committed:
Update Reservation.js

Add class name to the back button.

Wahaj-Ali committed:
Update Reservation.scss

Use media query to style the component for a responsive design.

Wahaj-Ali committed:
Update the the logic so when the user logins in then the app also dispatches 'fetchcurrentuser' action as well as the user id is now stored in the state after successful login.

Now the user can see 'my reservations' without first making a reservation.

Wahaj-Ali committed:
Remove linter errors

Wahaj-Ali committed:
Update MyReservations.js

Remove house id column from the list of all reservations.

Wahaj-Ali committed:
Update MyReservations.scss

Use media query to add styles for mobile view.

Wahaj-Ali committed:
Remove linter erros.

Wahaj-Ali committed:
Update addHouseSlice.js

Add delete house action to be dispatched from the delete house page.

Wahaj-Ali committed:
Delete deleteHouseSlice.js

Wahaj-Ali committed:
Update store.js

Wahaj-Ali committed:
Update deleteHouse.js

Remove unnecessary useEffect hooks & modify the code to dispatch the correct delete hosue action.

Wahaj-Ali committed:
Update deleteHouse.js

Wahaj-Ali committed:
Merge pull request #12 from Wahaj-Ali/responsive-design

Responsive design

Wahaj-Ali committed:
Merge branch 'dev' into delete-house

Wahaj-Ali committed:
Remove linter errors.

Wahaj-Ali committed:
Merge pull request #10 from Wahaj-Ali/delete-house

Delete house feature

Wahaj-Ali committed:
Update MyReservations.scss

Add min-height for the reservations div.

Meltrust commented on src/App.js:
```diff
@@ -0,0 +1,47 @@
+  </>
+);
+
+function App() {
```
- [ ]  Please, kindly refactor this and all other functions in your project to ES6 arrow functions, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

You can do a search in Vscode for the "function" word, and correct as needed.  Shouldn't take you more than 5-10 min so don't worry:

![image](https://github.com/Wahaj-Ali/homely-frontend/assets/57421823/75f07bd4-f666-4983-8dc1-9ddbe620344f)

Wahaj-Ali replied:
Thanks @Meltrust for your valuable suggestions. I will try my best to implement the changes.

Meltrust reviewed (changes requested):
_notice:  Your app did not work properly for me on Firefox.  Tested successfully on Chrome._

Hi @Wahaj-Ali,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Nice code organization ✔️ 
-  App is working well✔️ 
-  Good readme ✔️ 
-  Nice frontend ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

![You can do it](https://media.giphy.com/media/FemTGpEZeK27M5fKL4/giphy.gif)

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

Wahaj-Ali committed:
Update App.js

Use arrow function instead of 'function ' keyword.

Wahaj-Ali committed:
Update README.md

- Add future features.
- Add authors and their social media contacts.

Meltrust reviewed (approved):
#### Hi @Wahaj-Ali,


# **Wow, you did it 🎉**
![Brilliant](https://media.giphy.com/media/d31w24psGYeekCZy/giphy.gif)
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
![applause](https://media.giphy.com/media/srg19CG0cKMuI/giphy.gif)

### To highlight

- Functions are now ES6✔️ 
- Awesome job✔️ 




Cheers and Happy coding!👏👏👏


------

Wahaj-Ali commented:
Thank you @Meltrust for your encouraging remarks 😄

Wahaj-Ali committed:
Install babel-jest and forcing jest to import the commonjs axios build by adding it in package.json

Wahaj-Ali committed:
Create AddHouse.test.js

Add integration test for AddHouse component.

Wahaj-Ali committed:
Create AddHouse.test.js.snap

Wahaj-Ali committed:
Install redux-mock-store.

Wahaj-Ali committed:
Create Login.test.js

Add test for Login component.

Wahaj-Ali committed:
Create Register.test.js

Wahaj-Ali committed:
Create DeleteHouse.test.js

Wahaj-Ali committed:
Create Reservation.test.js

Wahaj-Ali committed:
Remove Liter errors.

Wahaj-Ali committed:
Merge pull request #14 from Wahaj-Ali/tests

Tests

Wahaj-Ali committed:
Update README.md

Wahaj-Ali committed:
Update README.md

Add screenshots of the initial stage of the Kanban board.

Wahaj-Ali committed:
Update README.md

Add total amount of team members.

Wahaj-Ali committed:
Update README.md

Add link to the deployed version.

Wahaj-Ali committed:
Update Houses.js

Add useEffect hook to fetch houses automatically(without reloading the app) whenever a new house is added by the user.

Wahaj-Ali committed:
Use the link to the deployed version of the backend to make http requests.

In this commit
- Make changes to the reducers to contain the URL for the deployed version of the server side.

Wahaj-Ali committed:
Merge pull request #15 from Wahaj-Ali/apk

Deploy the App(both front-end & back-end).

Wahaj-Ali committed:
Update README.md

belmeetmule committed:
add screen mock image

belmeetmule committed:
Update readme to show screen mock image

Wahaj-Ali committed:
Update SideBar.css

Add media query to modify footer styles for mobile view.

Wahaj-Ali committed:
Update Houses.scss

Modify style for the house image on the main-page for a better-looking UI.

Wahaj-Ali committed:
Update addHouseSlice.js

Remove syntax error preventing from successfully deleting a house.

Wahaj-Ali committed:
Update deleteHouse.js

The app shows a message informing the user whenever a house is deleted from the list.

Wahaj-Ali committed:
Update Reservation.js

Modify the text for reservations page.

Wahaj-Ali committed:
Update HouseDetails.js

Add authorization, only logged in users can access the reservations page(only signed in users can book their visit).

Wahaj-Ali committed:
Update HouseDetails.scss

Style the disabled button when the user in not signed in.

Wahaj-Ali committed:
Update HouseDetails.scss

Decrease font-sizes for the mobile view.

Wahaj-Ali committed:
Merge pull request #16 from Wahaj-Ali/screen-mock

Screen mock

Wahaj-Ali committed:
Update README.md

Add the project image to the readme file.

belmeetmule reviewed (approved):
Excellent Work !! 

Kindly, merge it to the main branch !!

Happy coding 👍
